dataland-frontend/tests/e2e/fixtures/FixtureUtils.ts

```diff
@@ -1,26 +1,32 @@
-import { CompanyInformation, CompanyReport } from "@clients/backend";
+import { CompanyInformation, CompanyReport, LksgData } from "@clients/backend";
 import { generateCompanyInformation } from "./CompanyFixtures";
+import { getRandomReportingPeriod } from "@e2e/fixtures/common/ReportingPeriodFixtures";
 
 export type ReferencedReports = { [key: string]: CompanyReport };
 
 export interface FixtureData<T> {
   companyInformation: CompanyInformation;
   t: T;
+  reportingPeriod: string;
 }
 
 /**
- * Randomly generates a fixture datasets consisting of a specified number of pairs of companies and framework datasets
+ * Randomly generates a fixture datasets consisting of a specified number of groups of company information datasets, framework datasets and
+ * a reporting period
  *
- * @param generator a generator that generates a random framework dataset
+ * @param frameworkDataGenerator a generator that generates a random framework dataset of type T
  * @param numElements the number of elements to generate
  * @returns a list of numElements pairs of randomly generated companies associated to randomly generated framework datasets
  */
-export function generateFixtureDataset<T>(generator: () => T, numElements: number): Array<FixtureData<T>> {
+export function generateFixtureDataset<T>(frameworkDataGenerator: () => T, numElements: number): Array<FixtureData<T>> {
   const fixtureDataset = [];
   for (let id = 1; id <= numElements; id++) {
+    const data = frameworkDataGenerator();
     fixtureDataset.push({
       companyInformation: generateCompanyInformation(),
-      t: generator(),
+      t: data,
+      reportingPeriod: (data as LksgData)?.social?.general?.dataDate?.substring(0, 4) ||
```

Comment:
This seems odd: the generator takes generic data types but the reportingPeriod always assumes LkSG. Am I missing something here?

Reply:
ah ok due to the ||, I overlooked that. You can leave it as is.

Reply:
Marc and I found a better solution for this. I have pushed it and you can review it.

You can optionally insert a `reportingPeriodGenerator()` now into the `generateFixtureDataset<T>()` method to control the way that the reporting periods are derived.
If you don't insert a `reportingPeriodGenerator()`, the `getRandomReportinPeriod()` method is used for this.